packages/@coorpacademy-player-services/src/analytics.js

```diff
@@ -1,17 +1,33 @@
+// @flow strict
+
+import type {Config, Progression, State} from '@coorpacademy/progression-engine';
+import type {Resource, ResourceType} from './types';
+
+// eslint-disable-next-line no-shadow
+declare var window: {|
+  dataLayer: Array<{
+    event: string,
+    mediaType?: string,
+    location?: string
+  }>
+|};
+
 // eslint-disable-next-line import/prefer-default-export
-export const sendViewedMediaAnalytics = (resource, location) => {
+export const sendViewedMediaAnalytics = (resource: Resource, location: string) => {
   window.dataLayer = window.dataLayer || [];
```

Comment:
Maybe we can inject the dataLayer as a parameter, which is an empty array by default